### PR TITLE
Implement SessionResetter interface

### PR DESCRIPTION
**Pull request by khous**
This interface allows the driver to tell database/sql when a connection needs to be discarded from the pool. 
These methods in this hierarchy end up calling db.conn which requests a connection from the pool and asks the driver if a reset is needed. thus flushing the connection pool.

### CODE REVIEW

**Review by mphillips-infoblox — APPROVED** [review]
Can we make resyncPeriod configurable in filewatcher.go?

**Comment by khous** [comment]
> Can we make resyncPeriod configurable in filewatcher.go?

Yeah, but why? Do you think the filesystem will need to be checked more often than two seconds? Not sure if this is a good thing to open up to users who are likely to misinterpret its usage.

**Comment by mphillips-infoblox** [comment]
That is true, but we may want to make it longer depending on how the secret updating strategy is configured. 
https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically

**Comment by khous** [comment]
> That is true, but we may want to make it longer depending on how the secret updating strategy is configured.
> https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically

`fsnotify` is getting file updates immediately. The resync is just a catchall in case something fails afaict.

**Comment by mphillips-infoblox** [comment]
ok, lets leave it for now

File: conn.go [hunk 1]
```diff
@@ -95,6 +96,14 @@ func (c *managedConn) IsValid() bool {
 	return s.IsValid()
 }
 
+func (c *managedConn) ResetSession(ctx context.Context) error {
+	if c.reset {
```
Comment (daniel-garcia) [inline conn.go:100]: This variable is inspected but it is set when a mutex is set; this is a race condition.

**Pull request merged**